Source/Core/DolphinQt2/MenuBar.cpp

```diff
@@ -271,3 +309,58 @@ void MenuBar::InstallWAD()
 
   result_dialog.exec();
 }
+
+void MenuBar::PerformOnlineUpdate(const std::string& region)
+{
```

Comment:
This file's currently mostly GUI setup and signal connecting. Could you move this progress dialog logic somewhere else and call into it?